### PR TITLE
Feature/eiszhs to excel

### PR DESCRIPTION
**Pull request by kpatrashanu**

### CODE REVIEW

File: Services/Eiszhs/Eiszhs.Reporter/Program.cs [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+                .ConfigureServices(ConfigureServices)
+                .UseConsoleLifetime()
+                .Build()
+                .RunAsync(cts.Token)
+                .ConfigureAwait(false);
+        }
+
+        private static void ConfigureServices(HostBuilderContext context, IServiceCollection sc)
+        {
+            sc.AddHostedService<EiszhsReporterHostedService>();
```
Comment (Anny777) [inline Services/Eiszhs/Eiszhs.Reporter/Program.cs:40]: Add Addresses client and eiszhs client registration

File: Services/Eiszhs/Eiszhs.Reporter/Services/EiszhsReporterHostedService.cs [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+    using Eiszhs.Client;
+    using Microsoft.Extensions.Hosting;
+    using Microsoft.Extensions.Logging;
+
+    /// <inheritdoc/>
+    public class EiszhsReporterHostedService : IHostedService
+    {
+        private readonly ILogger<EiszhsReporterHostedService> logger;
+        private readonly IInfoEiszhsClient client;
+        private readonly ExcelService excelService;
```
Comment (Anny777) [inline Services/Eiszhs/Eiszhs.Reporter/Services/EiszhsReporterHostedService.cs:19]: Why not interface?

File: Services/Eiszhs/Eiszhs.Client/IInfoEiszhsClient.cs [hunk 3]
```diff
@@ -27,5 +28,12 @@ public interface IInfoEiszhsClient
         /// <returns>List of strings (house id) sorting by desc.</returns>
         /// <param name="ct">The cancellation token.</param>
         Task<IEnumerable<string>> GetInfoEiszhs(CancellationToken ct);
+
+        /// <summary>
+        /// Gets list of eiszhs info for period.
+        /// </summary>
+        /// <returns>List of of eiszhs info for period.</returns>
+        /// <param name="ct">The cancellation token.</param>
+        Task<IEnumerable<InfoResidentialConstruction>> GetListInfoForPeriod(CancellationToken ct);
```
Comment (Anny777) [inline Services/Eiszhs/Eiszhs.Client/IInfoEiszhsClient.cs:37]: Maybe GetReport?

File: Services/Eiszhs/Eiszhs.Client/InfoEiszhsClient.cs [hunk 4]
```diff
@@ -62,6 +62,22 @@ public async Task<IEnumerable<string>> GetInfoEiszhs(CancellationToken ct)
...
+        {
+            var uri = this.options.MakeGetFilteredListUri($"['CreatedDate','>=','{DateTime.Now.AddYears(-1).ToString("o")}']");
+            await this.apiAuthenticationService.AuthenticateAsync(this.httpClientFactory, ct).ConfigureAwait(false);
+            var client = this.httpClientFactory.CreateClient(uri);
+            this.logger.LogInformation("Getting filtered list");
+            using var response = await Policy.Handle<HttpRequestException>()
+                .WaitAndRetryAsync(5, retryAttempt =>
+                {
+                    return TimeSpan.FromSeconds(Math.Pow(2, retryAttempt));
+                }).ExecuteAsync(() => client.GetAsync(uri)).ConfigureAwait(false);
```
Comment (Anny777) [inline Services/Eiszhs/Eiszhs.Client/InfoEiszhsClient.cs:76]: Pass the cancellation token

File: Services/Eiszhs/Eiszhs.Reporter/Services/ExcelService.cs [hunk 5]
```diff
@@ -0,0 +1,99 @@
...
+        /// <inheritdoc/>
+        public MemoryStream CreateStream(IEnumerable<InfoResidentialConstruction> info)
+        {
+            if (info == null)
+            {
+                throw new ArgumentNullException(nameof(info));
+            }
+
+            MemoryStream stream = new MemoryStream();
+#if debug
```
Comment (Anny777) [inline Services/Eiszhs/Eiszhs.Reporter/Services/ExcelService.cs:35]: Maybe without if debug. Leave on permanent.

**Review by Anny777 — CHANGES_REQUESTED** [review]

**Review by DamienVStone — CHANGES_REQUESTED** [review]

**Review by DamienVStone — APPROVED** [review]

**Review by Anny777 — APPROVED** [review]

**Pull request merged**